sdk/identity/src/token_credentials/imds_managed_identity_credentials.rs

```diff
@@ -97,6 +93,9 @@ impl TokenCredential for ImdsManagedIdentityCredential {
 
         let mut query_items = vec![("api-version", MSI_API_VERSION), ("resource", resource)];
 
+        let mut headers = HeaderMap::new();
+        headers.insert("Metadata", "true".parse().unwrap());
```

Comment:
This new `unwrap` isn't my favorite, but I'll be updating the error handling for this crate for #771 very soon.